AuthenticatorShared/Core/Vault/Repositories/AuthenticatorItemRepository.swift

```diff
@@ -246,23 +300,29 @@ extension DefaultAuthenticatorItemRepository: AuthenticatorItemRepository {
     }
 
     func itemListPublisher() async throws -> AsyncThrowingPublisher<AnyPublisher<[ItemListSection], Error>> {
-        try await authenticatorItemService.authenticatorItemsPublisher()
-            .asyncTryMap { items in
-                try await self.itemListSections(from: items)
-            }
-            .eraseToAnyPublisher()
-            .values
+        try await itemListSectionPublisher().values
     }
 
     func searchItemListPublisher(
         searchText: String
     ) async throws -> AsyncThrowingPublisher<AnyPublisher<[ItemListItem], Error>> {
-        try await searchPublisher(
-            searchText: searchText
-        ).asyncTryMap { items in
-            items.compactMap { item in
-                ItemListItem(authenticatorItemView: item, timeProvider: self.timeProvider)
+        try await itemListSectionPublisher().map { sections -> [ItemListItem] in
+            let query = searchText.trimmingCharacters(in: .whitespacesAndNewlines)
+                .lowercased()
+                .folding(options: .diacriticInsensitive, locale: .current)
+            let allItems = sections.flatMap(\.items)
+
+            var matchedItems: [ItemListItem] = []
+
+            allItems.forEach { item in
+                if item.name.lowercased()
+                    .folding(options: .diacriticInsensitive, locale: nil)
+                    .contains(query) {
+                    matchedItems.append(item)
+                }
             }
+
+            return matchedItems.sorted { $0.name.localizedStandardCompare($1.name) == .orderedAscending }
```

Comment:
🤔 Could just `allItems.filter(...).sorted { ... }` be used here instead of using the additional helper variable `matchedItems`?
```suggestion
            let allItems = sections.flatMap(\.items)
                .filter { item in
                    item.name.lowercased()
                        .folding(options: .diacriticInsensitive, locale: nil)
                        .contains(query)
                }
                .sorted { $0.name.localizedStandardCompare($1.name) == .orderedAscending }
```

Reply:
👍 I like this approach. I removed allItems as well since we can just return the result.